subsystems/mvdm/ntvdm/bios/vidbios.c

```diff
@@ -2942,6 +2942,295 @@ static VOID VidBiosPrintCharacter(CHAR Character, BYTE Attribute, BOOLEAN UseAtt
     VidBiosSetCursorPosition(Row, Column, Page);
 }
 
+static VOID VidBiosSetPixel(WORD X, WORD Y, BYTE Page, BYTE Value)
+{
+    SHORT ScreenWidth = VgaGetDisplayResolution().X;
+
+    if (IS_TEXT_MODE(Bda->VideoMode))
+    {
+        DPRINT1("Note, trying to modify pixel (%d,%d) @ page %d to %d in text mode %X - returning 0.\n", X, Y, Page, Value, Bda->VideoMode);
+        return;
+    }
+
+    switch (Bda->VideoMode)
+    {
+        case 0x04: /* CGA 320x200 4c */
+        case 0x05: /* CGA 320x200 4c */
+        case 0x06: /* CGA 640x200 2c  */
+        {
+            ULONG Address;
+            BYTE MemoryValue;
+            BOOL Xor;
+            BYTE MaxColor;
+            BYTE ValueShift;
+            BYTE PixelsPerByte;
+
+            if (Bda->VideoMode == 0x06)
+            {
+                MaxColor = 1;
+                ValueShift = 7 - X % 8;
+                PixelsPerByte = 8;
+            }
+            else
+            {
+                MaxColor = 3;
+                ValueShift = (3 - X % 4) * 2;
+                PixelsPerByte = 4;
+            }
+
+            Xor = (Value & 0x80) != 0;
+            Value &= MaxColor;
+            Address = TO_LINEAR((Y & 1) ? CGA_ODD_VIDEO_SEG : CGA_EVEN_VIDEO_SEG,
```

Comment:
As already done in other functions, you can put a
```c
WORD CgaSegment[] = { CGA_EVEN_VIDEO_SEG, CGA_ODD_VIDEO_SEG };
```
somewhere, and then here just do a `CgaSegment[Y & 1]`.